gtfs/spec/en/reference.md

```diff
@@ -472,8 +480,8 @@ Assigns stops from [stops.txt](#stopstxt) to areas.
 
 |  Field Name | Type | Presence | Description |
 |  ------ | ------ | ------ | ------ |
-| `area_id` | Foreign ID referencing `areas.area_id` | **Required** | Identifies an area to which one or multiple `stop_id`s belong. The same `stop_id` may be defined in many `area_id`s. |
-| `stop_id` | Foreign ID referencing `stops.stop_id` | **Required** | Identifies a stop. If a station (i.e. a stop with `stops.location_type=1`) is defined in this field, it is assumed that all of its platforms (i.e. all stops with `stops.location_type=0` that have this station defined as `stops.parent_station`) are part of the same area. This behavior can be overridden by assigning platforms to other areas. |
+| `area_id` | Foreign ID referencing `areas.area_id` | **Required** | Identifies an area to which one or multiple `stop_id`s belong. The same `stop_id` may be defined in many `area_id`s.<br><br>May also identify a group of stops and/or GeoJSON locations that together indicate locations where a rider may request pickup or drop off.<br><br>It is forbidden to define an `area_id` with the same value as a `stop_id` or `id` from `locations.geojson`. |
+| `stop_id` | Foreign ID referencing `stops.stop_id` or `id` from `locations.geojson` | **Required** | Identifies a stop or GeoJSON location. If a station (i.e. a stop with `stops.location_type=1`) is defined in this field, it is assumed that all of its platforms (i.e. all stops with `stops.location_type=0` that have this station defined as `stops.parent_station`) are part of the same area. This behavior can be overridden by assigning platforms to other areas. |
```

Comment:
I would be interested to hear about the experience of consumers already using these fields as they are currently designed. Where exactly does the impossibility lie?

Reply:
Ok, not impossible, but it sounds like a bad design or a hack. From the sentences "Foreign ID referencing `stops.stop_id`, `stop_areas.area_id`, or `id` from `locations.geojson` " and "It is forbidden to define an `area_id` with the same value as a `stop_id` or `id` from `locations.geojson`." these are the things come to my mind: 1. this is some afterthought. 2. Is this restriction also explained in stop_areas.area_id, stops.stop_id, locations.geojson? Almost, except that in locations.id only collision with stops is forbidden, but not with stop_areas.area_id (I guess it should be added there?)

Ah, just noticed now: the field is called stop_id. It's misleading. I'm agains having a field that is "foreign ID" to 3 different fields (I don't think this counts as a foreign id, but maybe I'm too conservative) but if everyone else agrees on this, then at least let's give it a proper name. It is NOT a stop_id, so let's not call it that. Maybe geographical_reference or stop_location? BTW the same goes for stop_areas.stop_id.

Reply:
> Almost, except that in locations.id only collision with stops is forbidden, but not with stop_areas.area_id (I guess it should be added there?)

Yes, good catch, it should be added. @tzujenchanmbd 

> It is NOT a stop_id, so let's not call it that. Maybe geographical_reference or stop_location?

The original iteration of Flex (v1) had a separate field in stop_times to refer only to polygons, but you still had to define `stop_times.stop_id` since it is a required field. Because of this, you had to use a "dummy" id in `stop_times.stop_id` which, the v2 drafting community decided, was worse than just having `stop_times.stop_id` able to reference ids from multiple files.

Reply:
But naming conventions are important. Can't the field name be changed to something less confusing?

Reply:
We cannot change `stop_times.stop_id`, as that is part of the core GTFS standard.

Reply:
To me it looks like a breaking change anyway, because until now stop_times.stop_id was referencing only stops.stop_id. If you allow anything else here (as this change does) then it's a breaking change, in which case it could also be renamed. Or if you want to be semi backwards compatible then you still could add a new field with some better name, and add to stop_id and the new field that they are conditionally forbidden and only one of them can be defined. This would enable current producers to continue to produce their feed with stop_id (with only referencing stop_ids) OR use the new field instead if they also want to reference areas and/or locations.

Reply:
I do not think it is a breaking change to add to what a field can reference. If you have established that A can reference X, there is no contradiction to later say A can *also* reference Y and Z.

> This would enable current producers to continue to produce their feed with stop_id (with only referencing stop_ids) OR use the new field instead if they also want to reference areas and/or locations.

I do not see how this is an advantage for producers.

Reply:
I accept Gavriel's point about this being a little difficult to model in a relational database but I don't believe that it must be the goal of the spec to make it so.

I also agree that using the name "stop_location" (which we do in our internal model) would have been preferable if this was part of the first iteration of GTFS ~15 years ago.

However, given that GTFS has no versions and pretty ironclad backwards-compatibility guarantees, I think using stop_id rather than introducing a new field is the least bad solution given the constraints that we face. For my part, I have been consuming the stop_id that refers to three different types of entities for years and it has not given me any issues that are worse than all the other problems that exist when you mix static and flexible transit data.

Having said that, I would be open to do a consensus finding exercise to figure out what the community thinks.

Lastly, I would like to emphasize that I don't want to dismiss Gavriel's points but rather argue that GTFS is in many ways highly pragmatic which _does_ lead these kinds of design decisions. On balance, I think I'm happier with this approach compared to what other standards follow which say that they follow more "robust" engineering principles.

Reply:
Re: [this comment](https://github.com/google/transit/pull/388#discussion_r1269923819)
Restriction added https://github.com/google/transit/pull/388/commits/04503319f90bf5a161057e09e72bb5e9fd76d594, thanks!